### ISSUE TITLE
Subsequent nixos-rebuild switch fails on wrongly modified nix.conf

### ISSUE THREAD

**Issue by Abdillah** [NixOS/nixpkgs#21576]
## Issue description
If `nix.conf` wrongly modified (syntax error, wrong value, etc) from `/etc/nixos/configuration.nix`, then the only way to revert is reboot and choose prior generation.

Do we need to prevent this before it's occurrence?

### Steps to reproduce
[Warning] Do not actually try this at home.
1. Change `/etc/nixos/configuration.nix` to include a wrong syntax, mine was (with additional semicolon)
```
nix.extraOptions = ''
    auto-optimise-store = true;
  '';
```
2. Cast `sudo nixos-rebuild switch`.
3. From now on, all nix command will complain about
```
error: configuration option ‘auto-optimise-store’ should be either ‘true’ or ‘false’, not ‘true;’
```

## Technical details
```
$ nixos-version                                                                                                                                                                   
16.09.1368.435b5f8 (Flounder)
```

**Comment by anderspapitto** [comment]
note: in this particular case, you might have been able to recover by passing `--option auto-optimise-store true` to your nix commands while trying to recover

**Comment by Abdillah** [comment]
While `nix-env` and friends able to receive `--option`, `nixos-rebuild switch` can't. I haven't find anyway to recover without `nixos-rebuild` (without reboot, of course). Thanks for the hint note.

**Comment by mmahut** [comment]
Are there any updates to this issue, please?

**Comment by Artturin** [comment]
fixed since 4a9d9928dc01421b878cfe4a16961c1e4862d846 which added a checkPhase
